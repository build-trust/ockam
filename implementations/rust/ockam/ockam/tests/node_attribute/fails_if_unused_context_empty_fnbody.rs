// This test checks that #[ockam::node] causes a compile time error
// if the function is passed a parameter of type `ockam::Context` but is unused.

#[ockam::node]
async fn main(_ctx: ockam::Context) {}
