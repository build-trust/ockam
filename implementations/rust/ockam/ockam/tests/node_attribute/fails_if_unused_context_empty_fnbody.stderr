error: Function body Cannot be Empty.
 --> $DIR/fails_if_unused_context_empty_fnbody.rs:5:10
  |
5 | async fn main(_ctx: ockam::Context) {}
  |          ^^^^

error[E0601]: `main` function not found in crate `$CRATE`
 --> $DIR/fails_if_unused_context_empty_fnbody.rs:4:1
  |
4 | / #[ockam::node]
5 | | async fn main(_ctx: ockam::Context) {}
  | |______________________________________^ consider adding a `main` function to `$DIR/tests/node_attribute/fails_if_unused_context_empty_fnbody.rs`
